xds/src/main/java/io/grpc/xds/GcpAuthenticationFilter.java

```diff
@@ -254,23 +262,42 @@ public void sendMessage(ReqT message) {}
 
   private static final class LruCache<K, V> {
 
-    private final Map<K, V> cache;
+    private Map<K, V> cache;
+    private int maxSize;
 
     LruCache(int maxSize) {
+      this.maxSize = maxSize;
       this.cache = new LinkedHashMap<K, V>(
           maxSize,
           0.75f,
           true) {
         @Override
         protected boolean removeEldestEntry(Map.Entry<K, V> eldest) {
-          return size() > maxSize;
+          return size() > LruCache.this.maxSize;
         }
       };
     }
 
     V getOrInsert(K key, Function<K, V> create) {
       return cache.computeIfAbsent(key, create);
     }
+
+    private void resizeCache(int newSize) {
+      if (newSize >= maxSize) {
+        maxSize = newSize;
+        return;
+      }
+      LinkedHashMap<K, V> newCache = new LinkedHashMap<>(newSize, 0.75f, true);
+      // Copy the MRU entries (which are at the end of access-order map)
+      List<Entry<K, V>> entries = new ArrayList<>(cache.entrySet());
+      int start = Math.max(0, entries.size() - newSize);
+      for (int i = entries.size() - 1; i >= start; i--) {
+        Map.Entry<K, V> entry = entries.get(i);
+        newCache.put(entry.getKey(), entry.getValue());
```

Comment:
Since this line is uncovered, the test checking the caching resizing actually has zero entries. That means it also doesn't attempt to confirm an entry is preserved in the cache.